### PR TITLE
Add a few corrections for the code management

### LINKED ISSUE #19: Change Main activity layout
When we start inserting emojis,  replace the header text with the emojis we inserted

### LINKED ISSUE #33: Allow emoji repitition in connection code
For the moment, it is impossible to have twice the same emoji in a connection code.

### PR DESCRIPTION
**Pull request by Stellucidam**

### CODE REVIEW

File: app/src/main/java/ch/heigvd/pro/b04/android/Home/EmojiCodeAdapter.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import androidx.lifecycle.LifecycleOwner;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import ch.heigvd.pro.b04.android.R;
+public class EmojiCodeAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+
+    private static final int VIEW_TYPE_EMOJI = 1;
```
Comment (glsubri) [inline app/src/main/java/ch/heigvd/pro/b04/android/Home/EmojiCodeAdapter.java]: If only one type, not needed

File: app/src/main/java/ch/heigvd/pro/b04/android/Home/EmojiCodeAdapter.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    @Override
+    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
+        ((EmojiViewHolder) holder).bindEmoji(
+                emojis.get(position)
+        );
+    }
+
+    @Override
+    public int getItemCount() {
+        System.out.println("Emojis size : " + emojis.size());
```
Comment (glsubri) [inline app/src/main/java/ch/heigvd/pro/b04/android/Home/EmojiCodeAdapter.java]: Use logcat

File: app/src/main/java/ch/heigvd/pro/b04/android/Home/EmojiCodeAdapter.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        return emojis.size();
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        return VIEW_TYPE_EMOJI;
+    }
+
```
Comment (glsubri) [inline app/src/main/java/ch/heigvd/pro/b04/android/Home/EmojiCodeAdapter.java]: Not needed

File: app/src/main/java/ch/heigvd/pro/b04/android/Home/Home.java [hunk 4]
```diff
@@ -34,13 +36,34 @@ public int getSpanSize(int position) {
...
 
-        EmojiAdapter emojiGridAdapter = new EmojiAdapter(state, this);
-        emojiGrid.setAdapter(emojiGridAdapter);
+        EmojiAdapter emojiAdapter = new EmojiAdapter(state, this);
+        emojiGrid.setAdapter(emojiAdapter);
         emojiGrid.setLayoutManager(manager);
 
+        // Code of selected emojis
+        RecyclerView emojiCode = findViewById(R.id.home_emoji_code);
+        GridLayoutManager llm = new GridLayoutManager(this, 4);
```
Comment (glsubri) [inline app/src/main/java/ch/heigvd/pro/b04/android/Home/Home.java]: Clarify name `llm`

File: app/src/main/java/ch/heigvd/pro/b04/android/Home/Home.java [hunk 5]
```diff
@@ -34,13 +36,34 @@ public int getSpanSize(int position) {
...
+        //llm.setOrientation(RecyclerView.HORIZONTAL);
+        manager.setSpanSizeLookup(new GridLayoutManager.SpanSizeLookup() {
+            @Override
+            public int getSpanSize(int position) {
+                return position == 0 ? 4 : 1;
```
Comment (glsubri) [inline app/src/main/java/ch/heigvd/pro/b04/android/Home/Home.java]: Factorize the `4` variable

File: app/src/main/java/ch/heigvd/pro/b04/android/Home/Home.java (hunk 4 above)
Comment (glsubri) [inline app/src/main/java/ch/heigvd/pro/b04/android/Home/Home.java]: Factorize the `4` variable

File: app/src/main/java/ch/heigvd/pro/b04/android/Home/Home.java [hunk 6]
```diff
@@ -34,13 +36,34 @@ public int getSpanSize(int position) {
...
+        emojiCode.setLayoutManager(llm);
+
         state.getToken().observe(this, token -> {
-            Intent intent = new Intent(this, PollActivity.class);
-            startActivity(intent);
+            if (token.getToken() == "Error") {
+                // TODO : Make it a bit prettier if possible, but at least, it does the job
+                Toast toast = Toast.makeText(this, "Wrong code !", Toast.LENGTH_LONG);
+                toast.setGravity(0, 0, 0);
+                toast.show();
```
Comment (glsubri) [inline app/src/main/java/ch/heigvd/pro/b04/android/Home/Home.java:61]: Doesn't work

File: app/src/main/java/ch/heigvd/pro/b04/android/Home/Home.java [hunk 7]
```diff
@@ -9,6 +9,7 @@
 import androidx.appcompat.app.AppCompatActivity;
 import androidx.lifecycle.ViewModelProvider;
 import androidx.recyclerview.widget.GridLayoutManager;
+import androidx.recyclerview.widget.LinearLayoutManager;
```
Comment (glsubri) [inline app/src/main/java/ch/heigvd/pro/b04/android/Home/Home.java]: Useless

File: app/src/main/java/ch/heigvd/pro/b04/android/Home/HomeViewModel.java [hunk 8]
```diff
@@ -60,13 +63,13 @@ public void addNewEmoji(Emoji emoji) {
         // TODO : Factorize this logic in a model class, and test it.
 
         List<Emoji> buffer = queue.getValue();
```
Comment (glsubri) [inline app/src/main/java/ch/heigvd/pro/b04/android/Home/HomeViewModel.java]: Remove `buffer`

**Review by glsubri — CHANGES_REQUESTED** [review]

File: app/src/main/java/ch/heigvd/pro/b04/android/Home/Home.java (hunk 6 above)
Comment (Stellucidam) [reply app/src/main/java/ch/heigvd/pro/b04/android/Home/Home.java:61]: It actually does, don't know why it doesn't on your end...

**Review by glsubri — APPROVED** [review]

**Review by Stellucidam — COMMENTED** [review]
We actually need it for Travis CI tests (commit #64e400b)

**Pull request merged**